addon/components/uni-form.js

```diff
@@ -29,7 +29,10 @@ export default Component.extend({
 
       this.set('isLoading', true);
 
-      this.get('onSubmit')().then(() => this.setIsLoading(false), () => this.setIsLoading(false));
+      let promise = this.get('onSubmit')();
+      if (promise && typeof promise.then === "function") {
```

Comment:
This is in case somebody (don't know why) returns an object with the "then" property, you'll have an exception on the next link when it tries to execute it.